tests/Stripe/HttpClient/CurlClientTest.php

```diff
@@ -352,4 +353,120 @@ public function testSetRequestStatusCallback()
             \Stripe\ApiRequestor::setHttpClient(null);
         }
     }
+
+    /**
+     * @after
+     */
+    public function tearDownTestServer()
+    {
+        $this->stopTestServer();
+    }
+
+    // This is a pretty flaky/uncertain way to try and get an
+    // http server to deliver the body in separate "chunks".
+    //
+    // It seems to work but feel free to just skip or delete
+    // if this flakes.
+    public function testExecuteRequestWithRetriesCallsWriteFunctionWithChunks()
+    {
+        $chunk1 = 'First, bytes';
+        $chunk2 = 'more bytes';
+        $chunk3 = 'final bytes';
+        $serverCode = <<<EOF
+<?php
```

Comment:
I'm trying to think through if there are better ways we can provide the code than a raw string... but haven't figured anything out yet. Maybe we can have the test code be in fixture files? Mainly I'm just worried that we might have some kind of syntax error or something in here that ends up hiding an issue. Not sure what would be better though...

Reply:
I added a step to run `php -l` on the raw strings - this should catch syntax errors.

Don't want to do the fixtures, as that involves separating the test logic in the server from the test logic in the test and makes things harder to read/change.

It's a bit better in the similar [stripe-node](https://github.com/stripe/stripe-node/blob/7e1b9f7c5a17e4fee00c1eee43e83f6a26c29a31/test/StripeResource.spec.js#L189) [test server](https://github.com/stripe/stripe-node/blob/9b58b95e2fbd0296ceb43aa6b1854785edbc3561/testUtils/index.js#L14) where you can control server and client from the same process. Don't know how to accomplish anything similar in PHP.